core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -168,6 +168,13 @@ class KafkaApis(val requestChannel: RequestChannel,
     val response =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+        // Clearing out the cache for groups that belong to an offsets topic partition for which this broker was the leader,
+        // since this broker is no more a replica for that offsets topic partition
```

Comment:
@MayureshGharat Thanks, this comment helps, but I was hoping we could elaborate a bit more on the specific partition reassignment case we're trying to protect.